### DISCUSSION TITLE
Fastest way to filter fields?

### DISCUSSION THREAD

**Discussion by SlothCroissant** [JoshClose/CsvHelper#1893] [Q&A]
First off, love the CsvHelper tool, it's been wildly useful for me in recent weeks!

I have a CSV file with ~15 million rows. The first column is an IP address, and the CSV file is filtered by A-Z on this column (meaning 0.x.x.x.x then 1.x.x.x then 2.x.x.x then 3.x.x.x.x and so on).

My goal is to parse/filter the CSV file by first octet of the IP Address. So 1.2.3.4 would be filtered as "1", and in my use case, I could pull out all IPs that start with "1."

I'd like to filter this by every octet (0..255) and I'm looking for the fastest way to do it. Currently, I can runa  GetRecords() and use ShouldSkipRecord to do the work, but I'm mostly curious to see if I can do it more efficiently/quickly.

Sample code that works now that I'd like to optimize a bit:

``` csharp
class Program
{
    static void Main(string[] args)
    {
        foreach (var i in Enumerable.Range(0, 255))
        {
            Console.WriteLine($"Processing octet {i}... ");

            var config = new CsvConfiguration(CultureInfo.InvariantCulture)
            {
                HasHeaderRecord = false,
                ShouldSkipRecord = (args) => !args.Record[0].StartsWith($"{i}.")
            };
            using (var streamreader = new StreamReader("mycsv.csv"))
            using (var csvreader = new CsvReader(streamreader, config))
            {
                var records = csvreader.GetRecords<Foo>().ToList();
            }
        }
    }
}

public class Foo
{
    [Index(0)]
    public string start { get; set; }

    [Index(1)]
    public string end { get; set; }

    [Index(2)]
    public string loc { get; set; }
}
\
```

**Comment by JoshClose** [comment]
`CsvReader` is a forward only reader, so using `ShouldSkipRecord` is probably the best way. If you need to do multiple things in a file, you would need some metadata about the file to be efficient. In that case you could scan the whole file once and create metadata that you could query.
